src/components/headers/NavigationStackHeader.tsx

```diff
@@ -0,0 +1,36 @@
+/*
```

Comment:
So we have 3 really similar Headers: NavigationBaseHeader, NavigationStackHeader, and StackHeader!
TBH I find this a bit confusing. Maybe we could find a way to make it dryer?

Reply:
As discussed on Slack, since the use-cases for each of those headers is quite different, I didn't find a way to make it DRYer than it already is. `NavigationBaseHeader` is using the `BaseHeader` and only has the minimal necessary code to make it work with the scroll context and navigation props. The `NavigationStackHeader` naturally needs to use the `NavigationStackHeader`, just as the `StackHeader` uses the `BaseHeader`.